### PR TITLE
Shortcuts: Open Note

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
Okay, so back on track after a bunch of setup PRs.  What we have here is the ability to open a note from a shortcut.  That should be fairly simple, but there were a lot of core technologies that needed to be added to do this.  So sorry but this PR is a little long.  One the plus side several hundred lines of it are all xml files that are auto generated and shouldn't need much review.

Anyways, so the shortcut that I have added here allows you to specify or search for a note and then open it in Simplenote.  Straight forward, but it is defining some of the core tech we will need for more advanced shortcuts.

Moves forward: #1145 
relies on #1162 

### Test
1. build and launch Simplenote.  Make sure you are logged in
2. Open the shortcuts app, create a new shortcut, find simplenote, select Open Note
3. In the shortcut that gets created select a note to open
4. Run the shortcut.  Simplenote should launch and open that note.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> These changes do not require release notes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: IntentsExtension/Tools/ExtensionCoreDataWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        guard FileManager.default.fileExists(atPath: StorageSettings().sharedStorageURL.path) else {
+            return nil
+        }
+        return extensionResultsController
+    }
+
+    func context() -> NSManagedObjectContext {
+        coreDataManager.managedObjectContext
+    }
+}
```
Comment (jleandroperez) [inline IntentsExtension/Tools/ExtensionCoreDataWrapper.swift:34]: This class feels a bit off:

1. `StorageSettings()` should probably be passed via parameters
2. `coreDataManager` will abort execution if the shared CD stack cannot be loaded
3. **BUT** `resultsController()` makes sure that `sharedStorageURL` exists, before invoking `coreDataManager`, so we're avoiding the abort

Most of what this class does could probably be handled by `CoreDataManager` itself, WDYT? Perhaps something like this?

```
let coreDataManager: CoreDataManager? = try? CoreDataManager(storageURL: ..., for:. intents)
```

And then, wherever you must build the ResultsController, try unwrapping the coreDataManager?

File: IntentsExtension/Models/Note+Widget.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// This file contains the required class structure to be able to fetch and use core data files in widgets and intents
```
Comment (jleandroperez) [inline IntentsExtension/Models/Note+Widget.swift]: Nipticky: Filename should probably be `Note+Intents` ?

File: IntentsExtension/Models/SPManagedObject+Widget.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// This file contains the required class structure to be able to fetch and use core data files in widgets and intents
```
Comment (jleandroperez) [inline IntentsExtension/Models/SPManagedObject+Widget.swift]: Nipticky: `SPManagedObject+Intents` ?

File: Simplenote/SimplenoteAppDelegate+Swift.swift [hunk 4]
```diff
@@ -295,13 +295,45 @@ extension SimplenoteAppDelegate {
...
+
+    private func handleShortcutActivity(_ userActivity: NSUserActivity) -> Bool {
+        guard let type = ActivityType(rawValue: userActivity.activityType),
+              simperium.user?.authenticated() == true else {
+            return false
+        }
+
+        switch type {
+        case .newNoteShortcut:
+            SimplenoteAppDelegate.shared().noteEditorViewController.createNote(from: nil)
```
Comment (jleandroperez) [inline Simplenote/SimplenoteAppDelegate+Swift.swift]:
```suggestion
            noteEditorViewController.createNote(from: nil)
```

**Review by jleandroperez — COMMENTED** [review]
✅ Verified that the Open Note shortcut effectively opens the target note

Sending you a couple questions!! TY!!

File: IntentsExtension/Tools/ExtensionCoreDataWrapper.swift (hunk 1 above)
Comment (charliescheer) [reply IntentsExtension/Tools/ExtensionCoreDataWrapper.swift:34]: Yeah, I was looking at that results controller a little skeptically.  Updated

File: Simplenote/SimplenoteAppDelegate+Swift.swift (hunk 4 above)
Comment (charliescheer) [reply Simplenote/SimplenoteAppDelegate+Swift.swift]: done in another PR

**Review by roundhill — APPROVED** [review]
This is so cool! Worked well for me.

What do you think about maybe focusing the note editor after opening the note? So say I have a "Shopping List" note, it'd be cool if it opened with the editor active so I could type ~~"beer"~~ "soda" straight away.

**Comment by charliescheer** [comment]
Good idea, TY.  Done!

**Pull request merged**